Dockerfile

```diff
@@ -6,21 +6,70 @@ ARG ARCH=amd64
 RUN dpkg --add-architecture ${ARCH} && \
     apt-get update && apt-get install -y \
     build-essential \
-    git wget \
-    debhelper devscripts \
-    pkg-config \
-    liblzma-dev:${ARCH} \
-    libssl-dev:${ARCH} \
-    libglib2.0-dev:${ARCH}
+    git wget curl \
+    debhelper devscripts
+
+RUN if [ "${ARCH}" != "armhf" ]; then \
+        apt-get install -y \
+        pkg-config \
+        liblzma-dev:${ARCH} \
+        libssl-dev:${ARCH} \
+        libglib2.0-dev:${ARCH}; \
+    fi
 
 # To provide support for Raspberry Pi Zero W a toolchain tuned for ARMv6 architecture must be used.
 # https://tracker.mender.io/browse/MEN-2399
+ENV ARMV6_TOOLCHAIN_ROOT="/armv6-eabihf--glibc--stable-2018.11-1"
 RUN if [ "${ARCH}" = "armhf" ]; then \
         wget -nc -q https://toolchains.bootlin.com/downloads/releases/toolchains/armv6-eabihf/tarballs/armv6-eabihf--glibc--stable-2018.11-1.tar.bz2 \
         && tar -xjf armv6-eabihf--glibc--stable-2018.11-1.tar.bz2 \
         && rm armv6-eabihf--glibc--stable-2018.11-1.tar.bz2; \
     fi
 
+# Get depdendencies from upstream, manually donwloading deb packages, and fake pkg-config
+RUN if [ "${ARCH}" = "armhf" ]; then \
+        set -e; \
+        ln -s /bin/true /usr/bin/pkg-config; \
```

Comment:
Maybe not convincing enough, but I wrote:
```
# NOTE: pkg-config is used from go build to obtain cflags and libs required to build C code; however
# for armhf we will explicitly pass these flags for go to use our custom sysroot instead of the system
# one. Alternatively pkg-config supports setting a prefix, but that would require patching the source code.
```

Reply:
:+1: